src/App.tsx

```diff
@@ -15,71 +15,100 @@ export const goods = [
-    <h1 className="title">No goods selected</h1>
+type State = {
+  selectedGood: string,
+};
 
-    <h1 className="title is-flex is-align-items-center">
-      Jam is selected
+export class App extends Component<{}, State> {
+  state = {
+    selectedGood: 'Jam',
+  };
 
-      {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
-      <button
-        data-cy="ClearButton"
-        type="button"
-        className="delete ml-3"
-      />
-    </h1>
+  className = 'has-background-success-light';
 
-    <table className="table">
-      <tbody>
-        <tr data-cy="Good">
-          <td>
-            <button
-              data-cy="AddButton"
-              type="button"
-              className="button"
-            >
-              +
-            </button>
-          </td>
+  selectedItem = (item: string) => {
+    this.setState({ selectedGood: item });
+  };
 
-          <td data-cy="GoodTitle" className="is-vcentered">
-            Dumplings
-          </td>
-        </tr>
+  clearItem = () => {
+    this.setState({ selectedGood: '' });
+  };
 
-        <tr data-cy="Good" className="has-background-success-light">
-          <td>
-            <button
-              data-cy="RemoveButton"
-              type="button"
-              className="button is-info"
-            >
-              -
-            </button>
-          </td>
+  render() {
+    const {
+      selectedGood,
+    } = this.state;
 
-          <td data-cy="GoodTitle" className="is-vcentered">
-            Jam
-          </td>
-        </tr>
+    return (
+      <main className="section container">
 
-        <tr data-cy="Good">
-          <td>
-            <button
-              data-cy="AddButton"
-              type="button"
-              className="button"
-            >
-              +
-            </button>
-          </td>
+        {
+          selectedGood === ''
+            ? <h1 className="title">No goods selected</h1>
+            : (
+              <h1 className="title is-flex is-align-items-center">
+                {selectedGood}
+                {' '}
+                is selected
+                {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
+                <button
+                  data-cy="ClearButton"
+                  type="button"
+                  className="delete ml-3"
+                  onClick={this.clearItem}
+                />
+              </h1>
+            )
+        }
 
-          <td data-cy="GoodTitle" className="is-vcentered">
-            Garlic
-          </td>
-        </tr>
-      </tbody>
-    </table>
-  </main>
-);
+        <table className="table">
+          <tbody>
+            {
+              goods.map(item => (
+                <tr
+                  data-cy="Good"
+                  key={item}
+                  className={
+                    selectedGood === item
+                      ? this.className
+                      : ''
+                  }
```

Comment:
Better install [classnames](https://www.npmjs.com/package/classnames) and use it instead of ternary operator.
The ternary operator is very limited in its functionality to be used in classNames. (this way usually did in real projects and would be great to work with it now and would be needed to do in test tasks).